### PR TITLE
New Serverless Pattern - Receive Email using Amazon SES

### PR DESCRIPTION
**Pull request by biswanathmukherjee**
*Issue #, if available:*

*Description of changes:*
This sample project demonstrates how to receive email using Amazon SES and process using AWS Lambda.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ses-lambda-sam/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+## How it works
+
+
+Please refer to the architecture diagram below:
+
+![End to End Architecture](images/architecture.png)
+
+* User sends email to the configured domain in Amazon SES.
+* Amazon SES receives the email and then applies the rule set to trigger an AWS Lambda function. 
+* The AWS Lambda functoin processes information from the SES event.
```
Comment (bfreiberg) [inline ses-lambda-sam/README.md:62]:
```suggestion
* The AWS Lambda function processes the information from the SES event.
```

File: ses-lambda-sam/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+## Testing
+
+
+1. From the command line, please execute the below command to tail logs of the AWS Lambda function. Please replace `FunctionName` from the `sam deploy` output and also your region. 
+   ```bash
+   aws logs tail --follow /aws/lambda/{FunctionName} --region {your-region}
+   ```
+
+
+2. Make a note of `ReceivingEmailAddress` from the `sam deploy` output and send email to that email id. Send an email to this Id.
```
Comment (bfreiberg) [inline ses-lambda-sam/README.md:73]:
```suggestion
2. Make a note of `ReceivingEmailAddress` from the `sam deploy` output and send email to that address.
```

File: ses-lambda-sam/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+
+3. You should see logline similar to following in the Lambda function log:
```
Comment (bfreiberg) [inline ses-lambda-sam/README.md:76]: Is this in near real-time or should the user expect some delay?

File: ses-lambda-sam/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
+{
+  "title": "Receive email by Amazon SES and process using AWS Lambda function",
```
Comment (bfreiberg) [inline ses-lambda-sam/example-pattern.json:2]:
```suggestion
  "title": "Receive email with Amazon SES and process them using an AWS Lambda function",
```

File: ses-lambda-sam/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  "description": "This sample project demonstrates how to receive email using Amazon SES and process using AWS Lambda.",
```
Comment (bfreiberg) [inline ses-lambda-sam/example-pattern.json:3]:
```suggestion
  "description": "This sample project demonstrates how to receive email from Amazon SES and process them with AWS Lambda.",
```

File: ses-lambda-sam/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  "language": "Python",
+  "level": "200",
+  "framework": "SAM",
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "User sends email to the configured domain in Amazon SES.",
+      "Amazon SES receives the email and then applies the rule set to trigger an AWS Lambda function.",
+      "The AWS Lambda functoin processes information from the SES event."
```
Comment (bfreiberg) [inline ses-lambda-sam/example-pattern.json:12]:
```suggestion
      "The AWS Lambda function processes the information from the SES event."
```

File: ses-lambda-sam/src/process-email.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import json
+
+# Create an AWS Lambda function that processes emails received by Amazon SES
+def lambda_handler(event, context):
+    # Get the email message from the event data
+    message = event['Records'][0]['ses']['mail']
```
Comment (bfreiberg) [inline ses-lambda-sam/src/process-email.py:6]: Can their be multiple records?

File: ses-lambda-sam/README.md (hunk 1 above)
Comment (biswanathmukherjee) [reply ses-lambda-sam/README.md]: Fixed.

File: ses-lambda-sam/README.md (hunk 3 above)
Comment (biswanathmukherjee) [reply ses-lambda-sam/README.md]: It is event-driven. For me it took <5 sec. So I have added line to expect a few seconds of delay.

File: ses-lambda-sam/example-pattern.json (hunk 4 above)
Comment (biswanathmukherjee) [reply ses-lambda-sam/example-pattern.json]: Fixed.

File: ses-lambda-sam/example-pattern.json (hunk 5 above)
Comment (biswanathmukherjee) [reply ses-lambda-sam/example-pattern.json]: Fixed.

File: ses-lambda-sam/src/process-email.py (hunk 7 above)
Comment (biswanathmukherjee) [reply ses-lambda-sam/src/process-email.py:6]: There will be one record as each email triggers the Lambda function.

File: ses-lambda-sam/example-pattern.json (hunk 6 above)
Comment (biswanathmukherjee) [reply ses-lambda-sam/example-pattern.json]: Fixed.

**Comment by bfreiberg** [comment]
Looks good, thanks for your contribution. A DA will merge your PR soon

**Pull request merged**